src/pybamm/util.py

```diff
@@ -102,38 +102,43 @@ def __getitem__(self, key):
                 f"'{key}' not found. Best matches are {best_matches}"
             ) from error
 
-    def search(self, key, print_values=False):
+    def search(self, keys, print_values=False):
         """
-        Search dictionary for keys containing 'key'. If print_values is True, then
-        both the keys and values will be printed. Otherwise just the values will
-        be printed. If no results are found, the best matches are printed.
+        Search dictionary for keys containing all terms in 'keys'.
+        If print_values is True, both the keys and values will be printed.
+        Otherwise, just the keys will be printed. If no results are found,
+        the best matches are printed.
```

Comment:
Let's add the inputs to the docstring as well:

```suggestion
        Search dictionary for keys containing all terms in 'keys'.
        If print_values is True, both the keys and values will be printed.
        Otherwise, just the keys will be printed. If no results are found,
        the best matches are printed.

        Parameters
        ----------
        keys : str or list of str
            Search term(s)

        print_values : bool, optional
            If True, print both keys and values. Otherwise, print only keys.
            Default is False.
```